espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt

```diff
@@ -199,9 +206,49 @@ class SourceDocument constructor(
         serializer?.endTag(NAMESPACE, tagName)
     }
 
+    private fun serializeComposeNode(semanticsNode: SemanticsNode?, depth: Int) {
+        if (semanticsNode == null) {
+            return
+        }
+        val nodeElement = ComposeNodeElement(semanticsNode)
+        serializer?.startTag(NAMESPACE, COMPOSE_TAG_NAME)
```

Comment:
cannot we retrieve the class name of the compose node?

Reply:
I tried, looks like they don't have any specific class as such

Reply:
I assume we could use https://developer.android.com/reference/kotlin/androidx/compose/ui/semantics/Role for this purpose

Reply:
Most of the time `SemanticsProperties.Role` is null. What should we do in such cases?

Reply:
Use the default value like we do now (e.g. `ComposeNode`)

Reply:
does that mean some nodes will be having Role as a class attribute and some of them will have no class attribute?

Reply:
I think `SemanticsProperties.Role` is more of an attribute of the node than the node tag name itself. 
Anyway, I am open to change it to anything you suggest

Reply:
For there that have role null simply set the default ComposeNode value. This is the same we do in UIA2, just setting it to android.view.View instead